opm/simulators/wells/GasLiftStage2_impl.hpp

```diff
@@ -665,11 +692,18 @@ removeSurplusALQ_(const Opm::Group &group,
             if (state.checkEcoGradient(well_name, eco_grad)) remove = true;
         }
         if (remove) {
-            const GradInfo &gi = this->dec_grads_.at(well_name);
-            state.updateRates(well_name, gi);
+            state.updateRates(well_name);
             state.addOrRemoveALQincrement( this->dec_grads_, well_name, /*add=*/false);
             recalculateGradientAndUpdateData_(
-                dec_grad_itr, /*increase=*/false, dec_grads, inc_grads);
+                        dec_grad_itr, /*increase=*/false, dec_grads, inc_grads);
+
+            // The dec_grads and inc_grads needs to be syncronized across ranks
+            // We first restrict the vectors to only wells on own rank
+            // before we gather all wells from all ranks
+            auto dec_grads_local = globalToLocalGradVector_(dec_grads);
+            auto inc_grads_local = globalToLocalGradVector_(inc_grads);
+            dec_grads = localToGlobalGradVector_(dec_grads_local);
+            inc_grads = localToGlobalGradVector_(inc_grads_local);
```

Comment:
Shouldn't we just do the extraction of the local grads in localToGlobalGradVector? Currently we are creating a vector (with pairs with a string as the first entry) in globalToLocalGradVector_, just to create another vector (with pairs with an int as the first entry).

Reply:
We might want to make sure that these functions do not do anything for a serial run. There is a search involved!